packages/twenty-front/src/modules/workflow/search-variables/utils/getVariableTag.ts

```diff
@@ -0,0 +1,50 @@
+import { Node } from '@tiptap/core';
+import { mergeAttributes, RawCommands } from '@tiptap/react';
+
+export const getVariableTag = () =>
+  Node.create({
+    name: 'variableTag',
+    group: 'inline',
+    inline: true,
+    atom: true,
+
+    addAttributes: () => ({
+      variable: {
+        default: null,
+      },
+    }),
+
+    renderHTML: ({ node, HTMLAttributes }) => {
+      const variable = node.attrs.variable as string;
+      const variableWithoutBrackets = variable.replace(
+        /\{\{([^}]+)\}\}/g,
+        (_, variable) => {
+          return variable;
+        },
+      );
+
+      const parts = variableWithoutBrackets.split('.');
+      const displayText = parts[parts.length - 1];
+
+      return [
+        'span',
+        mergeAttributes(HTMLAttributes, {
+          'data-type': 'variableTag',
+          class: 'variable-tag',
+        }),
+        displayText,
+      ];
+    },
+
+    // @ts-expect-error - addCommands is missing from the NodeConfig type
+    addCommands: () => ({
+      insertVariableTag:
+        (variable: string) =>
+        ({ commands }: { commands: Partial<RawCommands> }) => {
+          return commands.insertContent?.({
+            type: 'variableTag',
+            attrs: { variable },
+          });
+        },
```

Comment:
As I mentioned in another component, this function could be moved to a custom Extension to please TypeScript. Does it make sense, or am I missing something?

Reply:
Nope I did not manage to fix that part. My function `insertVariableTag` returns a callback that returns a `Command`. But TS expects my function to return a Command directly. While I am simply following the doc.